src/native-common/Modal.ts

```diff
@@ -16,10 +16,6 @@ import Types= require('../common/Types');
 
 export class Modal extends RX.Modal {
     isDisplayed(modalId?: string): boolean {
-        if (modalId === '') {
```

Comment:
@berickson1 I proposed to remove it for several resons:

1. With this `throw` we never go to else statement

```typescript
public isModalDisplayed(modalId?: string) : boolean {
        if (modalId) {
            return this._findIndexOfModal(modalId) !== -1;
        } else {
            if (this._overlayStack.length > 0) {
                const modals = _.filter(this._overlayStack, iter => (iter instanceof ModalStackContext));
                return modals.length > 0;
            }
            return false;
     }
}
```
2. When I call `isDisplayed()` instead of getting boolean `false|true` I get exception which looks weird, especially if there is logic which handles the case when we do not pass `modalId`.

Does that make sence? What do you think?

Reply:
I'm not sure that's correct. The throw only occurs when the string is empty. If undefined is passed to this function, the else path will be exercised.

With the new code, what happens if you try to create a modal with an ID that has an empty string?

Reply:
@berickson1 Based on your comment logic, for `Modal.isDisplayed: boolean` should be like this 

```typescript
 if (Modal.isDisplayed()) {} // User will get boolean result
```

```typescript
 if (Modal.isDisplayed('')) {} // User will get exeption
``` 

Is it correct? If yes, I'll revert changes. BTW, `Popup` does not page such exception, maybe we need to add it too?

Reply:
Yeah. I'm mostly pointing out the change in behaviour. I'd assume that there was a reason for the previous behaviour, which is mostly what I'm trying to understand
Based on the existing validation code in Modal.ts (https://github.com/Microsoft/reactxp/blob/master/src/native-common/Modal.ts). An empty string is never a valid popupId.

Creating a popup with empty string ID will throw

@erictraut Thoughts on the behaviour change here?